crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -598,8 +598,15 @@ impl CheckpointFragment {
             SuiError::from("Waypoint diff is not valid")
         );
 
-        // TODO:
-        // - check that the certs includes all missing certs on either side.
+        // Check that the fragment contains all missing certs indicated in diff.
+        for digests in [&self.diff.first.items, &self.diff.second.items].iter() {
+            for digest in digests.iter() {
```

Comment:
nit: Use iterator `chain()` would be cleaner

Reply:
Nifty, thanks!